target/scripts/helpers/postfix.sh

```diff
@@ -91,3 +91,32 @@ function _vhost_ldap_support() {
 #
 # /etc/aliases is handled by `alias.sh` and uses `postalias` to update the Postfix alias database. No need for `postmap`.
 # http://www.postfix.org/postalias.1.html
+
+# Add an key with an value to Postfix's main configuration file
+# or update an existing key. An already existing key can be updated
+# by either appending to the existing value (default) or by prepending.
+#
+# @param ${1} = key name in Postfix's main configuration file
+# @param ${2} = new value (appended or prepended)
+# @param ${3} = "append" (default) or "prepend" [OPTIONAL]
+function _add_to_or_update_postfix_main() {
+  local KEY=${1:?Key name is required}
+  local NEW_VALUE=${2:?New value is required}
+
+  if grep -q -E "^${KEY}" /etc/postfix/main.cf; then
+    KEY=$(_escape_for_sed "${KEY}")
+    NEW_VALUE=$(_escape_for_sed "${NEW_VALUE}")
+    local SED_STRING="/${NEW_VALUE}/! s|^(${KEY} *=.*)|\1 ${NEW_VALUE}|g"
+
+    if [[ ${3:-append} == 'prepend' ]]; then
+      SED_STRING="/${NEW_VALUE}/! s|^(${KEY}) *= *(.*)|\1 = ${NEW_VALUE} \2|g"
+    elif [[ ${3:-append} != 'append' ]]; then
+      _log 'err' "Action '${3}' in _add_to_or_update_postfix_main is unknown"
+      return 1
+    fi
+
+    sed -i -E "${SED_STRING}" /etc/postfix/main.cf
+  else
+    postconf "${KEY} = ${NEW_VALUE}"
+  fi
+}
```

Comment:
I still find the following easier to grok:
- All parameters are covered up at the top for the method
- The default `ACTION` is clear and much nicer than the conditonals that keep setting it as default, the `case` statement reads better IMO.
- Your log call is for `err` but [the method expects `error`](https://github.com/docker-mailserver/docker-mailserver/blob/bb2038e8c65eaeaef20c61694fa68041de0a8d25/target/scripts/helpers/log.sh#L63), the `return 1` is fine though 👍 

```suggestion
function _add_to_or_update_postfix_main() {
  local KEY=${1:?Key name is required}
  local NEW_VALUE=${2:?New value is required}
  local ACTION=${3:-append}

  # If entry does not exist, add it - otherwise update with ACTION:
  if ! grep -q -E "^${KEY}" /etc/postfix/main.cf; then
    postconf "${KEY} = ${NEW_VALUE}"
  else
    KEY=$(_escape_for_sed "${KEY}")
    NEW_VALUE=$(_escape_for_sed "${NEW_VALUE}")
    local SED_STRING

    case "${ACTION}" in
      ('append')
        SED_STRING="/${NEW_VALUE}/! s|^(${KEY} *=.*)|\1 ${NEW_VALUE}|g"
        ;;
      ('prepend')
        SED_STRING="/${NEW_VALUE}/! s|^(${KEY}) *= *(.*)|\1 = ${NEW_VALUE} \2|g"
        ;;
      (*)
        _log 'error' "Action '${3}' in _add_to_or_update_postfix_main is unknown"
        return 1
        ;;
    esac

    sed -i -E "${SED_STRING}" /etc/postfix/main.cf
  fi
}
```